Creating Module 3 Spatial Crosswalk Tables

wokenny13 opened the pull request:
There are 3 new TADA Module 3 functions focused on helping users to apply unique site-specific criteria for assessments.

If a user is only interested in a ML sites level of assessments, they only need to run these 3 functions in order:

1) TADA_CreateParamRef()
2) TADA_CreateUseParamRef()
3) TADA_CreateSpatialRef()

Now, if a user is interested in an AU level of assessments, they can run two **additional** 'helper' functions prior to running TADA_CreateSpatialRef(). **See TADA Module 3b vignette for more info to walk through the process**:

4) TADA_CreateUseAURef()
5) TADA_CreateWaterUseParamRef()

This PR also reflects some minor updates to TADA_CreateParamRef() and TADA_CreateUseParamRef().

The excel file options in the functions still need updates, but the workflow should work when excel = FALSE

wokenny13 committed:
Update CriteriaMethods.R

This creates DefineMagnitude in which is the dataframe users will define Magnitude Criteria components

wokenny13 committed:
DefineMagnitude Updates

Fixed some formatting with the DefineMagnitude output table and other Continued developments.

wokenny13 committed:
Updated DefineMagnitude()

DefineMagnitude format has been updated. Minor bug fix in TADA_CreateParamRef() was fixed.

wokenny13 committed:
updates from first mod 3 functions pr

wokenny13 committed:
Update CST.csv

wokenny13 committed:
MagnitudeSummary (roughdraft)

need to review logic in summary output and handling of empty column cases and how to best summarize impairments

wokenny13 committed:
Magnitude Excrusion summary

wokenny13 committed:
Merge branch 'Create-Module-3-Reference-Tables' into Create-Module-3-AURef

wokenny13 committed:
Update CriteriaMethods.R

wokenny13 committed:
Create CriteriaPlot.R

wokenny13 committed:
Delete CriteriaPlot.R

wokenny13 committed:
Create CriteriaPlot.R

wokenny13 committed:
Update from Mod 3 Current PR

wokenny13 committed:
Draft AU to Use Ref functions

Define Magnitude Excursion Summary

wokenny13 committed:
Updates to AU-ML-USE crosswalk

Summary of magnitude excursions edits with formatting and logic are also included in this commit.

wokenny13 committed:
Update TADAModule3.Rmd

wokenny13 committed:
Updates to Mod 3 workflows Vignette

wokenny13 committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
Update TADA_CreateUseParamRef

Renamed paramUseRef argument to useParamRef

wokenny13 committed:
Update Module 3 Functions

Removed duplicate TADA functions that were moved around from different branches.

wokenny13 committed:
Update ATTAINSCrosswalks

wokenny13 committed:
Minor Text Updates

wokenny13 committed:
Scratch Work TADA_CreateUseAURef

Adding WaterType column and consider renaming of function.

wokenny13 committed:
Update TADA_CreateUseAURef

New required argument inputs. WaterType column will be included in this output. ATTAINS.ParameterName column added as well.

Users will need to specify unique spatial criteria for a Param-Use-WaterType during this function.

wokenny13 committed:
Minor Text Changes

wokenny13 committed:
Rename and Update TADA_CreateWaterUseParamRef

wokenny13 committed:
Updates

wokenny13 committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
Updates to Geospatial function

chunk size and blank space in URL handling

wokenny13 committed:
Update Geospatial function

wokenny13 committed:
Update ATTAINSCrosswalks.R

Editing documentation

Updates to 2 Spatial crosswalks:
TADA_CreateUseAURef
TADA_CreateWaterUseParamRef

wokenny13 committed:
Updates to TADA_CreateSpatialRef

committing current progress with updates to this function.

wokenny13 committed:
Edits to TADA Spatial

Mod3b vignette and function updates

wokenny13 committed:
Update TADAModule3b.Rmd

wokenny13 committed:
Updates to the Spatial Crosswalk Tables

Removed CriteriaMethods.R for now. This file will generate the final DefineCriteriaMethods table for users to fill out.

Documentation updates, TADA Module3b vignette updates, and other styling push

wokenny13 committed:
Created Example data frame

Data_Nutrients_UT_GetATTAINS contains output of our Data_Nutrients_UT when we run TADA_GetATTAINS.

wokenny13 committed:
Update All TADA Ref Files

Ran TADA_Update functions to update ref tables

wokenny13 committed:
Capitalization Argument Input Fix

lowercase 1st letter of an argument input

wokenny13 committed:
Documentation Update

wokenny13 committed:
DESCRIPTION and 3b Vignette Update

Return table outputs in Mod 3b vignette to view in vignette

Added rExpertQuery Github to DESCRIPTION.

wokenny13 committed:
Update DESCRIPTION

Typo fix

wokenny13 committed:
Update DESCRIPTION

wokenny13 committed:
Loading Example Data Issue

Attempted to store an example data of TADA_GetATTAINS(Data_Nutrients_UT) in the EPATADA Package.

Ran into an issue with updating example data.

For now, I will run the TADA_GetATTAINS() directly in the example data frame for the R documentation examples. Runtime may be around 5 minutes.

wokenny13 committed:
Updates

hillarymarler committed:
update wordlist and global vars

hillarymarler committed:
fix data set name issue

hillarymarler committed:
ex data set fixes

hillarymarler committed:
Add TADA's rExpertQuery API key

hillarymarler committed:
update ex data

wokenny13 committed:
Function documentation

added and removed param argument documentation that was missing

global variables fix - added missing ones to utilities. Note: I need to consider which column names to keep in the return data frame of each Mod 3 functions.

Mod 3 vignette update.

wokenny13 committed:
Updates

wokenny13 committed:
vignette update and mod 3 function updates

cleaned up code, made mod 3 functions more consistent with formatting and coding logic.

made org_id a required input.

Handling of cases of making the crosswalk tables reusable.

vignette workflow incorporated.

wokenny13 committed:
mod 3 Reusable Crosswalk table input Updates

Update logic on how to flag/identify new, existing and dropped row combinations of values and how to label them.

wokenny13 committed:
updates to mod 3 functions

IncludeOrExclude columns for ref tables

wokenny13 committed:
Reorganize (arrange) rows in outputs

make include rows show up on top.

reformat crosswalk table outputs.

mod 3 vignette modification

wokenny13 committed:
text edits

wokenny13 committed:
Document updates and spell checks

document updates

mod 3b vignette updates

spell checks/wordlist updates

wokenny13 committed:
Update TADAModule3b.Rmd

changed argument inputs in vignette to excel =FALSE

wokenny13 committed:
Merge develop branch

wokenny13 committed:
conflict fix in GeospatialFunctions.R

wokenny13 committed:
Required Maintainenance Run

Edit conflicts and ran maintenance

wokenny13 committed:
Update Global Variables

update global variables list and removed intermediate variables

Created/updated Data_Nutrients_GetATTAINS.rda

wokenny13 committed:
Mod 3 function updates

Improved TADA_ParamRef flagging logic.

Partial push of DefineCriteriaMethodology

wokenny13 committed:
Excluding TADA_DefineMagnitude

Still in development, commenting out for now

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

cristinamullin committed:
Merge branch 'geospatialrmd' into Create-Module-3-AURef

wokenny13 committed:
Update Utilities.R

Missing comma in global variables list

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
Temporary push for 5/12 Subgroup

Reflects TADA_DefineMagnitude (in development still - r documentation is still needed as well as some additional review and fixes to this function)

wokenny13 committed:
Push TADA_DefineMagnitude function

Updated documentation for TADA_DefineMagnitude function - ready for review.

Additional excel formatting updates to TADA_CreateSpatialRef

cristinamullin commented on vignettes/TADAModule3_PartA.Rmd:
````diff
@@ -344,13 +344,13 @@ Users may choose to voluntarily share their reference tables for their
 
 ```{r}
-NCTC_ParamUseRef3 <- TADA_CreateUseParamRef(
+NCTC_UseParamRef3 <- TADA_CreateUseParamRef(
````
This includes two rows for pH that are equivalent except for the TADA.ComparableDataIdentifier. This creates additional work since the use must be assigned to both even though they are both already assigned to the ATTAINS parameter "PH". Do we need TADA.ComparableDataIdentifier here if the ATTAINS parameter is already assigned as being the same for both?

wokenny13 replied:
I feel like there was a reason why TADA_ComparableDataIdentifier was left on here in TADA_CreateUseParamRef. But can't say for sure @hillarymarler do you recall any scenario in which TADA_ComparableDataIdentifier could be needed in TADA_CreateUseParamRef()?

I am unsure if there would ever be a case in which an ATTAINS parameter name, ex. if an organization is using ATTAINS Parameter name "Nitrogen", but it is only applied to a single use name for each TADA_ComparableDataIdentifier **"NITRATE + NITRITE_INORGANIC_AS NO3_MG/L", "NITRATE + NITRITE_UNFILTERED_AS N_MG/L", "TOTAL KJELDAHL NITROGEN (ORGANIC N & NH3)_UNFILTERED_AS N_MG/L"**
 whereas for all other use_names Nitrogen is assessed for only **"TOTAL KJELDAHL NITROGEN (ORGANIC N & NH3)_UNFILTERED_AS N_MG/L" and "NITRATE + NITRITE_INORGANIC_AS NO3_MG/L", "NITRATE + NITRITE_UNFILTERED_AS N_MG/L"**. 
![image](https://github.com/user-attachments/assets/dbbb66df-0cf2-4aa9-9db7-fadb1e939cee)

However if TADA_ComparableDataIdentifier is needed in TADA_CreateUseParamRef, then would it be needed in TADA_DefineMagnitude?

wokenny13 committed:
Updated example data frames

cristinamullin committed:
update TADAModule3a

cristinamullin commented on R/ATTAINSCrosswalks.R:
```diff
@@ -1239,24 +1292,24 @@ TADA_CreateUseParamRef <- function(.data, org_id = NULL, paramRef = NULL,
     warning(paste0(
-      "TADA_CreateParamRef: ",
-      "One or more organization identifiers entered by user is not found in your paramRef argument input.",
+      "TADA_CreateUseParamRef: ",
```
I wonder if this should be a "stop" instead of a "warning"? Because users cannot generate the ref for orgs missing from paramRef

wokenny13 replied:
I can change this to "stop"

wokenny13 committed:
CreateParam and UseParam updates

Removing EPA304a.PollutantName column.

wokenny13 committed:
Remove addition of EPA304a standards

Focusing on crosswalk ATTAINS and WQP for now.

wokenny13 committed:
Updates to mod 3 function

standardized column names when pulling in ATTAINS data and creating Mod 3 Crosswalk reference tables

updated vignettes

updated flag logic

wokenny13 committed:
Update test-Utilities.R

wokenny13 committed:
Renamed and Updates to Mod 3A vignette

wokenny13 committed:
updates

wokenny13 committed:
Updates to TADA_CreateParamRef and TADA_CreateUseParamRef

argument inputs created to assist with auto populating entries for exact matches.

allows for further specification if an org only wants org specific exact matches

wokenny13 committed:
Updates to TADA_CreateParamRef and TADA_CreateUseParamRef

wokenny13 committed:
Spatial Ref update

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
Updates to TADA_SpatialRef

wokenny13 committed:
Update documentation

For TADA_CreateParamRef and TADA_CreateUseParamRef

wokenny13 committed:
Merge branch 'Create-Module-3-AURef' of https://github.com/USEPA/EPATADA into Create-Module-3-AURef

wokenny13 committed:
Update ATTAINSCrosswalks.R

Updated documentation

wokenny13 committed:
function name change, bug fix and documentation updates

Updates to reflect changes to TADA_CreateParamRef()

documentation updates, bug fixes, and changed TADA_DefineMagnitude to TADA_DefineCriteriaMethodology

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

cristinamullin committed:
maintenance and TADA_GetATTAINSParamUseOrgRef bug fix

remove commented out code no longer needed with rExpertQuery update. TADA_UpdateATTAINSParamUseOrgRef is working now. Added TADA_UpdateATTAINSParameterWQPCharRef to required maintenance file. Updated all ref files and example data. Updated spelling list. Ran styler

cristinamullin committed:
Update CriteriaMethods.R

cristinamullin committed:
Update TADA_DefineCriteriaMethodology.Rd

cristinamullin commented on R/CriteriaMethods.R:
```diff
@@ -0,0 +1,436 @@
+#' crosswalk to assign any unique spatial criteria to a parameter, use, waterbody
+#' or monitoring site/assessment unit.
+#' 
+#' @param ref "TADA"
```
Can you add a definition for the ref = "TADA" input and an example that uses it?

wokenny13 committed:
Update to TADA_CreateParamRef and Removal of a TADA_DefineCriteriaMethodology arg input

Was not capturing the entire Index of allowable values when excel file was generated with TADA_CreateParamRef

Removed an arg input in TADA_DefineCriteriaMethodology for the time being.

wokenny13 committed:
Update ATTAINSCrosswalks.R

Excel CreateParamRef and CreateUseParamRef Flagging function update to reflect correct MATCH columns appropriately.
